### PR TITLE
Update in Curl recv message logic and default port values

### PR DESCRIPTION
**Pull request by tajnisha**
Signed-off-by: tajnisha <tajunnisha.n@intel.com>

### CODE REVIEW

File: lib/fdo.c [hunk 1]
```diff
@@ -1542,9 +1552,18 @@ static bool _STATE_TO2(void)
 			g_fdo_data->current_rvto2addrentry = g_fdo_data->current_rvto2addrentry->next;
 
 		}
+		if (0 == port) {
```
Comment (v-vasavi) [inline lib/fdo.c:1555]: Why are we setting this value in multiple places?

File: network/network_if_linux.c [hunk 2]
```diff
@@ -571,6 +571,105 @@ int32_t fdo_con_disconnect(fdo_con_handle handle)
...
+ * @retval true if header is valid and complete and false otherwise.
+ */
+bool has_header(char *buf,
+		size_t header_start_offset)
+{
+	char tmp[REST_MAX_MSGHDR_SIZE];
+	size_t cur_offset = header_start_offset;
+	bool ret = false;
+	for (;;) {
+		if (memset_s(tmp, sizeof(tmp), 0) != 0) {
```
Comment (v-vasavi) [inline network/network_if_linux.c:588]: Would the behavior of memset vary based on any platform properties?

File: network/network_if_linux.c [hunk 3]
```diff
@@ -571,6 +571,105 @@ int32_t fdo_con_disconnect(fdo_con_handle handle)
...
+			LOG(LOG_ERROR, "Memset() failed!\n");
+			goto err;
+		}
+		if (!read_until_new_line(tmp, REST_MAX_MSGHDR_SIZE, buf, &cur_offset)) {
+			goto err;
+		}
+
+		// end of header
+		if ((header_start_offset < cur_offset) && (tmp[0] == get_rest_hdr_body_separator())) {
```
Comment (v-vasavi) [inline network/network_if_linux.c:597]: should it be < or <= ?

File: network/network_if_linux.c (hunk 2 above)
Comment (tajnisha) [reply network/network_if_linux.c:588]: As far as I am aware, it behaves the same on all platforms.

File: network/network_if_linux.c (hunk 3 above)
Comment (tajnisha) [reply network/network_if_linux.c:597]: It is <.

File: lib/fdo.c (hunk 1 above)
Comment (tajnisha) [reply lib/fdo.c:1555]: One is during To1 for RV port and To2 for Owner port.

**Review by shrikant1407 — APPROVED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

**Review by v-vasavi — APPROVED** [review]

**Pull request merged**